rpc/src/rewards_recorder_service.rs

```diff
@@ -38,11 +42,25 @@ impl RewardsRecorderService {
                     if exit.load(Ordering::Relaxed) {
                         break;
                     }
-                    if let Err(RecvTimeoutError::Disconnected) = Self::write_rewards(
-                        &rewards_receiver,
-                        &max_complete_rewards_slot,
-                        &blockstore,
-                    ) {
+
+                    let rewards = match rewards_receiver.try_recv() {
```

Comment:
is this channel actually heavily contended? I'm thinking it could probably just be a recv, but this won't hurt

Reply:
> is this channel actually heavily contended?

No, single producer (`ReplayStage`) / single consumer (this service)

> I'm thinking it could probably just be a recv, but this won't hurt

I considered this; the `send()` will wake sleeping thread regardless:
https://github.com/crossbeam-rs/crossbeam/blob/aba6f562d877be1c5b48df1b30afd3164a4c4464/crossbeam-channel/src/flavors/list.rs#L306-L307

But `try_recv()`will avoid the spin-looping and yielding that happens in `backoff.snooze()`:
https://github.com/crossbeam-rs/crossbeam/blob/aba6f562d877be1c5b48df1b30afd3164a4c4464/crossbeam-channel/src/flavors/list.rs#L457-L474